src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -1999,28 +1956,88 @@ private CallResult ExecuteCall<TTracingInstructions>(EvmState vmState, byte[]? p
 
                         break;
                     }
-                case Instruction.BEGINSUB:
+                case Instruction.BEGINSUB | Instruction.TLOAD:
                     {
-                        if (!spec.SubroutinesEnabled) goto InvalidInstruction;
+                        if (spec.TransientStorageEnabled)
+                        {
 
-                        // why do we even need the cost of it?
-                        if (!UpdateGas(GasCostOf.Base, ref gasAvailable)) goto OutOfGas;
+                            Metrics.TloadOpcode++;
+                            var gasCost = GasCostOf.TLoad;
+
+                            if (!UpdateGas(gasCost, ref gasAvailable)) goto OutOfGas;
+
+                            stack.PopUInt256(out UInt256 storageIndex);
+                            storageCell = new(env.ExecutingAccount, storageIndex);
+
+                            byte[] value = _state.GetTransientState(storageCell);
+                            stack.PushBytes(value);
+
+                            if (typeof(TTracingStorage) == typeof(IsTracing))
+                            {
+                                _txTracer.LoadOperationTransientStorage(storageCell.Address, storageIndex, value);
+                            }
+
+                            break;
+                        }
+                        else
+                        {
+                            if (!spec.SubroutinesEnabled) goto InvalidInstruction;
+
+                            // why do we even need the cost of it?
+                            if (!UpdateGas(GasCostOf.Base, ref gasAvailable)) goto OutOfGas;
+
+                            goto InvalidSubroutineEntry;
+                        }
 
-                        goto InvalidSubroutineEntry;
                     }
                 case Instruction.RETURNSUB:
```

Comment:
or TSTORE

Reply:
wont this line only be reached if SubroutinesEip is enabled and TransientStorageOpcodes are disabled?